src/ott/tools/plot.py

```diff
@@ -424,3 +439,513 @@ def animate(
+          dt * (product_points[1, :, 1] - product_points[0, :, 1]),
+          angles="xy",
+          scale_units="xy",
+          scale=1,
+          **dict_pk["arrows_ifm"]
+      )
+
+  # End of static frame
+
+  ax.set_title(title)
+  ax.legend()
+  ax.grid(True)
+  ax.set_aspect("equal")
+  ax.set_xlim(*xlimits)
+  ax.set_ylim(*ylimits)
+
+  # Dynamic frame
+
+  # Initialize dynamic points at t=0
+  dyn_points_t = dyn_points
+
+  def update_frame(frame):
+
+    global dyn_points_t, v_points
+    t = times[frame]
+
+    # Update grid arrows (locations stay fixed)
+    if n_grid > 0 and t < 1.0:
+      dt = delta_times[frame - 1] if frame > 0 else delta_times[0]
+
+      times_t = jnp.ones((points_grid.shape[0],)) * t
+
+      if velocity_field is None:
+        v_grid = vel_brenier(times_t, points_grid)
+      else:
+        v_grid = velocity_field(times_t, points_grid)
+
+      quiver_grid.set_UVC(dt * v_grid[:, 0], dt * v_grid[:, 1])
+
+    # Update moving point arrows (locations move with time)
+    if plot_transport_arrows and not plot_monge:
+      dt = delta_times[frame - 1] if frame > 0 else delta_times[0]
+      if t >= 1.0:
+        # Stop displaying arrows at t=1.0
+        v_points = np.zeros_like(v_points)
+        if velocity_field is None:
+          dyn_points_t = dyn_end_points
+      else:
+        if velocity_field is not None:
+          v_points = velocity_field(
+              jnp.ones((dyn_points_t.shape[0],)) * t, dyn_points_t
+          )
+        else:
+          # velocity field is constant, path can be reconstructed.
+          dyn_points_t = (1 - t) * dyn_points + t * dyn_end_points
+
+      quiver_points.set_offsets(dyn_points_t)
+      quiver_points.set_UVC(dt * v_points[:, 0], dt * v_points[:, 1])
+      scatter_interm_points_after.set_offsets(dyn_points_t + v_points * dt)
+      scatter_interm_points_before.set_offsets(dyn_points_t)
+      # Make move for next iteration if integrating along path.
+      if velocity_field is not None:
+        dyn_points_t = dyn_points_t + dt * v_points
+
+    if (n_grid > 0 or plot_transport_arrows) and not plot_monge:
+      ax.set_title(title + " at time " + f"{t:.2f}")
+
+    if plot_ifm_interpolant:
+      product_points_at_t = (1 - t) * product_points[
+          0, :, :] + t * product_points[1, :, :]
+      prod_scatter.set_offsets(product_points_at_t)
+
+      if plot_ifm_arrows:
+        prod_quiver.set_offsets(product_points_at_t)
+      ax.set_title(title + " at time " + f"{t:.2f}")
+
+  ani = animation.FuncAnimation(
+      fig, update_frame, frames=n_frames, blit=False, interval=300, repeat=True
+  )
+  if n_frames == 1:
+    if save_prefix:
+      plt.savefig(save_prefix + title + ".pdf")
+    plt.show()
+  else:
+    plt.close()
+    if use_mp4:
+      display.display(display.HTML(ani.to_html5_video()))
+      if save_prefix:
+        ani.save(save_prefix + title + ".mp4", bitrate=2000)
+    else:
+      html = display.HTML(ani.to_jshtml())
```

Comment:
This logic is notebook-specific, I think we should handle in the notebook, not here.

Reply:
The user will have access to the animation since we return it anyways.